### ISSUE TITLE
SA1515 false positive on switch expressions

### ISSUE THREAD

**Issue by perlun** [DotNetAnalyzers/StyleCopAnalyzers#3392]
I stumbled upon the `SA1515` error (`Single-line comment should be preceded by blank line`) on the following code:

```c#
BigInteger value = numberBase switch
{
    Base.DECIMAL =>
        BigInteger.Parse(source[(start + startOffset)..current], numberStyles),

    Base.BINARY =>
        Convert.ToUInt64(numberCharacters, 2),

    Base.HEXADECIMAL =>
        // Quoting from
        // https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.parse?view=net-5.0#System_Numerics_BigInteger_Parse_System_ReadOnlySpan_System_Char__System_Globalization_NumberStyles_System_IFormatProvider_
        // If value is a hexadecimal string, the Parse(String, NumberStyles) method interprets value as a
        // negative number stored by using two's complement representation if its first two hexadecimal
        // digits are greater than or equal to 0x80. In other words, the method interprets the highest-order
        // bit of the first byte in value as the sign bit. To make sure that a hexadecimal string is
        // correctly interpreted as a positive number, the first digit in value must have a value of zero.
        //
        // We presume that all hexadecimals should be treated as positive numbers for now.
        BigInteger.Parse('0' + numberCharacters, numberStyles),

    _ =>
        throw new InvalidOperationException($"Base {(int)numberBase} not supported")
};
```

I noted that there have been some other issues with SA1515 for C# 7.0 switch _statements_ but note that this is for a switch _expression_ here. Could this be a bug/unsupported feature in StyleCop? :thinking: Using 1.2.0-beta.321 right now.

**Comment by sharwell** [comment]
One workaround seems to be:

```
    // Quoting from
    // https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.parse?view=net-5.0#System_Numerics_BigInteger_Parse_System_ReadOnlySpan_System_Char__System_Globalization_NumberStyles_System_IFormatProvider_
    // If value is a hexadecimal string, the Parse(String, NumberStyles) method interprets value as a
    // negative number stored by using two's complement representation if its first two hexadecimal
    // digits are greater than or equal to 0x80. In other words, the method interprets the highest-order
    // bit of the first byte in value as the sign bit. To make sure that a hexadecimal string is
    // correctly interpreted as a positive number, the first digit in value must have a value of zero.
    //
    // We presume that all hexadecimals should be treated as positive numbers for now.
    Base.HEXADECIMAL =>
        BigInteger.Parse('0' + numberCharacters, numberStyles),
```
